### ISSUE TITLE
RuntimeError: Non-empty 4D data tensor expected but got a tensor with sizes [1, 0, 1152, 1152]

### ISSUE THREAD

**Issue by mehran66** [ultralytics/ultralytics#1073]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Here is an error I get when training a segmentation model using DDP. I have not tested on one GPU but given I have had various issues with DDP, I thought maybe this is related to DDP as well.

File "/home/ubuntu/.config/Ultralytics/DDP/_temp_qo3qf6pr140061986570928.py", line 6, in <module>
    trainer.train()
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/ultralytics/yolo/engine/trainer.py", line 187, in train
    self._do_train(RANK, world_size)
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/ultralytics/yolo/engine/trainer.py", line 306, in _do_train
    self.loss, self.loss_items = self.criterion(preds, batch)
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/ultralytics/yolo/v8/segment/train.py", line 43, in criterion
    return self.compute_loss(preds, batch)
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/ultralytics/yolo/v8/segment/train.py", line 91, in __call__
    masks = F.interpolate(masks[None], (mask_h, mask_w), mode="nearest")[0]
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/torch/nn/functional.py", line 3922, in interpolate
    return torch._C._nn.upsample_nearest2d(input, output_size, scale_factors)
RuntimeError: Non-empty 4D data tensor expected but got a tensor with sizes [1, 0, 1152, 1152]
WARNING:torch.distributed.elastic.multiprocessing.api:Sending process 50792 closing signal SIGTERM
WARNING:torch.distributed.elastic.multiprocessing.api:Sending process 50793 closing signal SIGTERM
WARNING:torch.distributed.elastic.multiprocessing.api:Sending process 50794 closing signal SIGTERM
WARNING:torch.distributed.elastic.multiprocessing.api:Sending process 50795 closing signal SIGTERM
WARNING:torch.distributed.elastic.multiprocessing.api:Sending process 50796 closing signal SIGTERM
WARNING:torch.distributed.elastic.multiprocessing.api:Sending process 50797 closing signal SIGTERM
WARNING:torch.distributed.elastic.multiprocessing.api:Sending process 50798 closing signal SIGTERM
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 7 (pid: 50799) of binary: /home/ubuntu/mambaforge/bin/python
Traceback (most recent call last):
  File "/home/ubuntu/mambaforge/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/ubuntu/mambaforge/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/torch/distributed/run.py", line 766, in <module>
    main()
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 346, in wrapper
    return f(*args, **kwargs)
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/torch/distributed/run.py", line 762, in main
    run(args)
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/torch/distributed/run.py", line 753, in run
    elastic_launch(
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/torch/distributed/launcher/api.py", line 132, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/home/ubuntu/mambaforge/lib/python3.10/site-packages/torch/distributed/launcher/api.py", line 246, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError:

**Comment by Laughing-q** [comment]
@mehran66 hi I just tried the segmentation training with the following command which works good to me.
```bash
yolo segment train data=coco128.yaml model=yolov8l-seg.pt device=\'0,1\' imgsz=640 batch=16
```
can you also provide the command you used?

**Comment by Laughing-q** [comment]
@mehran66 hi I've reproduced you issue and it only happens when `overlap_mask=True` + `mask_ratio != 4` + there're no objects in one batch. And it also happens on single-gpu so I think it's not a DDP-specific issue. 
I'll make a PR to fix it later today. :)

**Comment by mehran66** [comment]
Thanks @Laughing-q yes, by trial and error,  I did come to the conclusion that the issue is related to overlap_mask=False + mask_ratio=1 as well. Thanks for fixing this!

**Comment by Laughing-q** [comment]
@mehran66 hey I've made a PR #1093 to fix this. But I noticed that you might encounter the case that there're no objects in one batch while training, which would lead a `unused gradient` error when you're training with DDP mode.
Please uncomment the following comments if you encounter that error:
[https://github.com/ultralytics/ultralytics/blob/840bc689f7f4497dcc670078af65adf9165ef4f7/ultralytics/yolo/v8/segment/train.py#L123-L127](https://github.com/ultralytics/ultralytics/blob/840bc689f7f4497dcc670078af65adf9165ef4f7/ultralytics/yolo/v8/segment/train.py#L123-L127)
Actually the `no objects in one batch` case rarely happens and the solution code looks a little odd so we just leave the code commented there. :)

**Comment by glenn-jocher** [comment]
@mehran66 fix should be merged to main in #1088 and new 8.0.44 release later today. Please let us know if this resolves your issue, thank you!

**Comment by Laughing-q** [comment]
@mehran66 closing the issue as it's solved, feel free to reopen it if you have any related issues. :)
